### PR TITLE
Add coverage for missing xms-error-responses test server scenarios

### PR DESCRIPTION
**Pull request by sarangan12**
This PR is related to issue https://github.com/Azure/autorest.typescript/issues/749. 

Added code to support isError and then added the test cases. 

@joheredi Please review and approve

Note: I have tested this code changes with locally built core-http changes. I will update this PR once we have merged and released this PR - https://github.com/Azure/azure-sdk-for-js/pull/11841. (Until then build will fail for this PR)

### CODE REVIEW

File: src/generators/operationGenerator.ts [hunk 1]
```diff
@@ -216,11 +217,12 @@ function buildResponses({ responses }: OperationSpecDetails): string[] {
   let parsedResponses: string[] = [];
   responseCodes.forEach(code => {
     // Check whether we have an actual mapper or a string reference
-    const { bodyMapper, headersMapper } = responses[code];
+    const { bodyMapper, headersMapper, isError } = responses[code];
     const bodyMapperString = buildMapper(bodyMapper, "bodyMapper");
     const headersMapperString = buildMapper(headersMapper, "headersMapper");
+    const isMapperString = isError ? `isError: ${isError}` : "";
```
Comment (joheredi) [inline src/generators/operationGenerator.ts:223]: What do you think about renaming this to `isError` or `isErrorMapperString`?

File: src/generators/operationGenerator.ts [hunk 2]
```diff
@@ -579,7 +581,7 @@ function getOperationParameterSignatures(
 function findLastRequiredParamIndex(
   params: ParameterWithDescription[]
 ): number {
-  for (let i = params.length; i--;) {
+  for (let i = params.length; i--; ) {
```
Comment (joheredi) [inline src/generators/operationGenerator.ts:584]: Would it be possible to revert formatting changes like this one?

**Review by joheredi — APPROVED** [review]
Looks great!

Please make sure to get a green CI before merging

File: src/generators/operationGenerator.ts (hunk 1 above)
Comment (sarangan12) [reply src/generators/operationGenerator.ts:223]: changed it to isErrorMapperString

File: src/generators/operationGenerator.ts (hunk 2 above)
Comment (sarangan12) [reply src/generators/operationGenerator.ts:584]: Will be careful next time.

**Pull request merged**